tests/chart_tests/test_resources.py

```diff
@@ -0,0 +1,44 @@
+import jmespath
+import pytest
+
+import tests.chart_tests as chart_tests
+from tests.chart_tests.helm_template_generator import render_chart
+
+
+def init_test_pod_resources():
+    chart_values = chart_tests.get_all_features()
+
+    kubernetes_objects = {
+        "StatefulSet": "spec.template.spec.containers",
+        "Deployment": "spec.template.spec.containers",
+        "CronJob": "spec.jobTemplate.spec.template.spec.containers",
+        "Job": "spec.template.spec.containers",
+        "DaemonSet": "spec.template.spec.containers",
+        "Pod": "spec.containers",
+    }
+
+    docs = render_chart(values=chart_values)
+
+    pod_docs = []
+    for key, val in kubernetes_objects.items():
+        pod_docs += jmespath.search(
+            f"[?kind == `{key}`].{{name: metadata.name, chart: metadata.labels.chart, kind: kind, container: {val}}}",
+            docs,
+        )
+
+    return {f'{doc["chart"]}_{doc["kind"]}_{doc["name"]}': doc["container"] for doc in pod_docs}
+
+
+test_pod_resources_configs_data = init_test_pod_resources()
```

Comment:
The function init_test_pod_resources  is run during import time since it's outside any function. This may unnecessarily slow down test initialization.

My suggestion would be to  move test_pod_resources_configs_data = init_test_pod_resources() inside the test class TestPodResources as part of a setup function.

Reply:
We can't move it inside the class because the result of this function is used to parametrize the class.